3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,75 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  const response = await fetch(url);
+  if (response.ok) {
+    return response.json();
+  }
+  throw new Error('Data request failed!');
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+  const url = 'https://pokeapi.co/api/v2/pokemon?limit=151';
+  try {
+    const response = await fetchData(url);
+    const pokemons = response.results;
+
+    // creating & populating 'select' with pokemon names
+    const select = document.createElement('select');
+    document.body.appendChild(select);
+
+    pokemons.forEach((pokemon) => {
+      const option = document.createElement('option');
+      option.value = pokemon.url;
+      option.textContent = pokemon.name;
+      select.appendChild(option);
+    });
+
+    // calling the function to fetch&render the image of the selected pokemon by passing its url
+    const handleSelection = (event) => {
+      fetchImage(event.target.value);
+
+      //removing previous image if it exist
+      const previousImage = document.body.querySelector('img');
+      if (previousImage) {
+        previousImage.remove();
+      }
+    };
+
+    //eventListener to trigger fetching the url after selecting pokemon
+    select.addEventListener('change', handleSelection);
+  } catch (error) {
+    renderError(error);
+  }
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchImage(url) {
+  try {
+    const response = await fetchData(url);
+    // populating the image tag
+    const image = document.createElement('img');
```

Comment:
Reuse the image element if it already exists

```suggestion
    // check for an existing image 
    let image = document.querySelector('img');
    // if it does not exist, create a new one
    if (!image) {
        image = document.createElement('img');
        document.body.appendChild(image);
     }
```